docs/docs/forms.mdx

```diff
@@ -142,44 +140,16 @@ write stories or rules for this case.
 
 ### Slot Mappings
 
-Rasa Open Source comes with four predefined mappings to fill the slots of a form
-based on the latest user message. Please see
-[Custom Slot Mappings](./forms.mdx#custom-slot-mappings) if you need a custom function
-to extract the required information.
-
-#### from_entity
-
-The `from_entity` mapping fills slots based on extracted entities.
-It will look for an entity called `entity_name` to fill a slot `slot_name`.
-If `intent_name` is `None`, the slot will be filled regardless of intent name.
-Otherwise, the slot will only be filled if the user's intent is `intent_name`.
+:::caution Changed in 3.0
+As of 3.0, [slot mappings](./domain.mdx#slot-mappings) are defined in the `slots` section of the domain.
+This change allows the same slot mapping to be reused across multiple forms, removing any unnecessary duplication.
+Please follow the [migration guide](./migration-guide.mdx#slot-mappings) to update your assistant.
+:::
 
-If `role_name` and/or `group_name` are provided, the role/group
-label of the entity also needs to match the given values. The slot mapping will not
-apply if the intent of the message is `excluded_intent`. Note that you can
-also define lists of intents for the parameters `intent` and `not_intent`.
+#### Unique entity mapping matching
 
-```yaml-rasa
-entities:
-- entity_name
-slots:
-  slot_name:
-    type: any
-    mappings:
-    - type: from_entity
-      entity: entity_name
-      role: role_name
-      group: group name
-      intent: intent_name
-      not_intent: excluded_intent
-forms:
-  your_form:
-    required_slots:
-        - slot_name
-```
-
-In `from_entity` mapping, when an extracted entity uniquely maps onto a slot,
-the slot will be filled even if this slot wasn't requested by the form.
+In `from_entity` mapping, when the attributes of an extracted entity (i.e, entity name, role, group) uniquely map
+onto a slot, the slot will be filled even if this slot wasn't requested by the form.
```

Comment:
isn't this the case for all slot mappings? They will apply even if the slot was not requested excepting conditions 🤔
This concept was kinda confusing in 2.x too, but now it's more so because slot filling happens somewhere else.

Is this only applicable when `store_entities_as_slots` is on? I think I need to understand this behaviour before suggesting anything 😅

Reply:
This is actually separate from auto-fill (which required `store_entities_as_slots`). 

To maintain regression of this unique entity mapping feature in forms, the new slot mappings maintains it during an active_loop with slot mappings that have conditions. So it will not extract the slot if the user utterance contained an entity which is mapped onto a non-unique `from_entity` mapping and the slot was not requested by the form.

Reply:
Let me see if I've got this right:
1. The definition of "unique entity mapping":

There is only one slot mapping in the entire domain which is `from_entity`  for a given (entity + attributes)  combo . 

2. Applicability of `from_entity` mappings when a form is active

`from_entity` mappings will get applied while a form is active iff either 
(1) the form requested the slot OR
(2) There is only one slot in the entire domain with a slot mapping for the specific (entity +attributes) combo listed.
 
Is this correct?

Reply:
I remember the point of this now (also based on example below), which was to prevent filling a slot with an entity intended to fill only the requested slot e.g. given a form that collects both `amount` (from `number` entity) and `account_number` (also from `number` entity), the form shouldn't go filling both slots with the first `number` entity extracted, it should leave the not-yet-requested one empty and wait for the user's response to the specific prompt.

If my understanding above is correct, I'll draft a short explanation of what "unique" means, because I'm sure someone besides me is gonna wonder about it

Reply:
> Applicability of from_entity mappings when a form is active

There is a nuance here when the form is active:
- either a slot is requested -> any of its mappings (unique or non-unique) become eligible for filling that slot.
- a slot was not requested but the user utterance has an entity that could fill that slot -> that entity can fill that slot only according to a unique entity mapping; otherwise a non-unique entity mapping won't apply and the slot remains unfilled.

Reply:
Right, I think we're on the same page now 😅 Let me make a phrasing suggestion then.

Reply:
Hmm, it's intended to allow other slots that are not in that form's required_slots list to be filled, if those slots don't have mapping with conditions (we assumed that any slot without conditions is meant to be filled by any user utterance outside forms too).

> it will still fill slots required by other forms with an extracted entity even if a form is active and the mapping is not unique across all slots in the domain.

Let's say you have slot_2 that should only be set by form_2, slot_2 should have its mapping with conditions to prevent this behaviour you mention in form_1 that doesn't require slot_2.

Maybe something else I need to clarify in the docs in `mapping conditions` section?

Reply:
I think this is the right place to explain it, will include in update I'm pushing